sources/web/datalab/polymer/modules/SettingsManager.ts

```diff
@@ -67,6 +71,40 @@ class SettingsManager {
   }
 
   /**
+   * Loads and executes the config file as specifed in appSettings.configUrl.
+   * This sets values on window.datalab. Resolves immediately if window.datalab is already set.
+   */
+  public static loadConfigToWindowDatalab():Promise<void> {
+    if (window.datalab) {
+      return Promise.resolve(); // Already loaded
+    }
+    if (window.datalab === undefined) {
+      window.datalab = {};
+    }
+    return SettingsManager.getAppSettingsAsync()
+      .then((settings: common.AppSettings) => {
+        if (settings.configUrl) {
+          return new Promise((resolve, reject) => {
+            const configScript = document.createElement('script');
+            configScript.src = settings.configUrl;
+            configScript.onload = () => {
+              resolve();
+            };
+            configScript.onerror = () => {
+              reject(new Error('Failed to load config file ' + settings.configUrl));
+            };
+            if (!document.body.getAttribute('data-version-id')) {
+              document.body.setAttribute('data-version-id', '');
+            }
+            document.head.appendChild(configScript);
+          }) as Promise<void>;
+        } else {
+          throw new MissingClientIdError();
```

Comment:
This should throw a different error message, maybe 'Cannot find config file setting'.